pandas/core/groupby/generic.py

```diff
@@ -1515,11 +1520,34 @@ def _wrap_agged_manager(self, mgr: Manager2D) -> DataFrame:
 
         if self.axis == 1:
             result = result.T
+            result = self._maybe_recast_columns_if_transposed(result)
 
         # Note: we only need to pass datetime=True in order to get numeric
         #  values converted
         return self._reindex_output(result)._convert(datetime=True)
 
+    def _maybe_recast_columns_if_transposed(self, result) -> DataFrame:
```

Comment:
umm what is this?

Reply:
Tried to explain in https://github.com/pandas-dev/pandas/pull/43213#issuecomment-917615165

Reply:
Instead of reverting #41706 (which IMO not a good option).

Another approach to fix this would be to amend `numeric_only`, so that `resolve_numeric_only` would be `False`.

That would be a smaller change, and only caveat being output dtype wouldn't exactly match `1.2.5` (which wouldn't be very bad if you ask me, since the dtype in `1.2.5` is also questionable).

Reply:
right we want the minimal change posssible for 1.3.x; even if the types are questionable it makes sense to try to replicate as much as possible.

Reply:
Made some changes. 
Although this returns `int64, int64, int64` instead of `int64, float64, int64` (in `1.2.5`). This maybe a better candidate for 1.3.x.
Let me know what you think.
```python
In [2]: import pandas as pd
   ...: 
   ...: print(pd.__version__)
   ...: df = pd.DataFrame([[1, 2, 3, 4, 5, 6]] * 3)
   ...: df.columns = pd.MultiIndex.from_product([["a", "b"], ["i", "j", "k"]])
   ...: for col in df.columns:
   ...:     if col[-1] == "j":
   ...:         df[col] = df[col].astype("Int64")
   ...: print("mixed int:")
   ...: result = df.groupby(level=1, axis=1).sum()
   ...: print(result)
   ...: print(result.dtypes)
   ...: 
1.4.0.dev0+727.ga37f1f982e
mixed int:
   i  j  k
0  5  7  9
1  5  7  9
2  5  7  9
i    int64
j    int64
k    int64
dtype: object